make/autoconf/jdk-version.m4

```diff
@@ -544,6 +549,9 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
       build version. Defaults to the build number.])])
   if test "x$with_macosx_bundle_build_version" = xyes; then
     AC_MSG_ERROR([--with-macosx-bundle-build-version must have a value])
+  elif test "x$with_macosx_bundle_build_version" = xno; then
+    # Interpret --without-* as empty string instead of the literal "no"
+    MACOSX_BUNDLE_BUILD_VERSION=
```

Comment:
Same objection here. And even if it works right now, if we do allow this, it means that all future changes in the build system that involves MACOSX_BUNDLE_BUILD_VERSION needs to be verified that it works with the empty string.

Reply:
From what I can tell, wouldn't MACOSX_BUNDLE_BUILD_VERSION be an empty string anyway if both VERSION_BUILD and VERSION_OPT = ""?

Reply:
That might be. If that is so, then you've most likely spotted a bug. In practice, you're either building an official build and have a build number, or an adhoc build with an opt string, so I'm not suprised noone has run into this.

Reply:
Both version build and version opt have valid --without options though, if both are set that could be problematic, since the issue wouldn't be stemming from a bug in that case. I'll double check just to be sure

Reply:
Ok, now I've tried to dig through Apple's somewhat confusing documentation on CFBundleVersion and CFBundleShortVersionString. My takeaway is that we should either publish the build number there, or 0 if we have no build number.

This is basically what is done by the code, and that was *exactly* what was done by the code, prior to the fix were we allowed an empty build number (JDK-8274980). So this code should also have been updated then. I've opened JDK-8282948 to fix this.

But still the effect for this PR is that a valueless --with-macosx-bundle-build-version is not accepted.

Reply:
Got it, should I link JDK-8282948 into this PR while addressing this issue in the meantime?
(On a side note the macOS tier 1 tests are also failing after this change, so you may be right that the build will inevitably fail when this has no value)

Reply:
JDK-8282948 will go in before this. But you can fix your code in the meantime to do the same yes/no check.

Reply:
alright